src/Features/CSharp/Portable/AssignOutParameters/AssignOutParametersAtStartCodeFixProvider.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.AssignOutParameters
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class AssignOutParametersAtStartCodeFixProvider : AbstractAssignOutParametersCodeFixProvider
+    {
+        protected override void TryRegisterFix(CodeFixContext context, Document document, SyntaxNode container, SyntaxNode location)
+        {
+            // Don't offer if we're already the starting statement of the container. This case will
+            // be handled by the AssignOutParametersAboveReturnCodeFixProvider class.
```

Comment:
💡 It seems like this comment belongs with the second `if` statement

Reply:
it was basically describing both.  if we're already at the start, no need to do anything.